ethcore/src/builtin.rs

```diff
@@ -260,10 +260,18 @@ impl Impl for ModexpImpl {
 
 		// calculate modexp: exponentiation by squaring.
 		fn modexp(mut base: BigUint, mut exp: BigUint, modulus: BigUint) -> BigUint {
-			if base == BigUint::zero() || modulus <= BigUint::one() { return BigUint::zero() }
+			match (base == BigUint::zero(), exp == BigUint::zero()) {
+				(_, true) => return BigUint::one(), // n^0 % m
+				(true, false) => return BigUint::zero(), // 0^n % m, n>0
+				(false, false) if modulus <= BigUint::one() => return BigUint::zero(), // a^b % 1 = 0.
+				_ => {}
+			}
 
 			let mut result = BigUint::one();
 			base = base % &modulus;
```

Comment:
At the very least you have to check if the modulus is zero beforehand or the code will panic.